packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1690,16 +1689,567 @@ async function prerenderToStream(
+         * Once we have settled all cache reads we restart the render and abort after a single Task.
+         *
+         * Unlike with the non PPR case we can't synchronously abort the render when a dynamic API is used
+         * during the initial render because we need to ensure all caches can be filled as part of the initial Task
+         * and a synchronous abort might prevent us from filling all caches.
+         *
+         * Once the render is complete we allow the SSR render to finish and use a combination of the popstoned state
+         * and the reactServerIsDynamic value to determine how to treat the resulting render
+         */
+
+        const PRERENDER_COMPLETE = 'NEXT_PRERENDER_COMPLETE'
+        const abortReason = getPostponedReason('NEXT_PRERENDER_COMPLETE')
+
+        let reactServerIsDynamic = false
+        function onError(err: unknown, errorInfo: ErrorInfo) {
+          if (isRenderInterruptedError(err)) {
+            reactServerIsDynamic = true
+            return
+          }
+
+          return serverComponentsErrorHandler(err, errorInfo)
+        }
+
+        const cacheSignal = new CacheSignal()
+        const prospectiveRenderPrerenderStore = {
+          cacheSignal,
+          // During the prospective render we don't want to synchronously abort on dynamic access
+          // because it could prevent us from discovering all caches in siblings. So we omit the controller
+          // from the prerender store this time.
+          controller: null,
+          // With PPR during Prerender we don't need to track individual dynamic reasons
+          // becuase we will always do a final render after caches have filled and we
+          // will track it again there
+          dynamicTracking: null,
+        }
+
+        let flightController = new AbortController()
+        // We're not going to use the result of this render because the only time we could potentially
+        // to that is if it completes in a microtask and that's likely very rare for any non-trivial app
+        const firstAttemptRSCPayload = await getRSCPayload(
+          tree,
+          ctx,
+          asNotFound
+        )
+        prerenderAsyncStorage.run(
+          // The store to scope
+          prospectiveRenderPrerenderStore,
+          // The function to run
+          ComponentMod.renderToReadableStream,
+          // ... the arguments for the function to run
+          firstAttemptRSCPayload,
+          clientReferenceManifest.clientModules,
+          {
+            nonce: ctx.nonce,
+            onError,
+            // we don't care to track postpones during the prospective render because we need
+            // to always do a final render anyway
+            signal: flightController.signal,
+          }
+        )
+
+        // When this resolves the cache has no inflight reads and we can ascertain the dynamic outcome
+        await cacheSignal.cacheReady()
+        flightController.abort(abortReason)
```

Comment:
maybe a silly question but why are we aborting here if the prospective render didn't even pass the controller into the render store?